olive/model/hf/hf_model.py

```diff
@@ -0,0 +1,51 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+from typing import Any, Callable, Dict, List, Optional, Union
+
+import torch
+from pydantic import validator
+
+from olive.common.config_utils import ConfigBase
+from olive.model.model_config import IOConfig
+
+
+class HFComponent(ConfigBase):
+    name: str
+    io_config: Union[IOConfig, str, Dict[str, Any]]
+    component_func: Union[str, Callable]
+    dummy_inputs_func: Union[str, Callable]
+
+
+class HFConfig(ConfigBase):
+    model_name: str = None
+    task: str = None
+    feature: str = "default"
+    # TODO: remove model_class and only use task
+    model_class: str = None
+    use_custom_implementation: bool = False
+    components: List[HFComponent] = None
+    config: Dict[str, Any] = None
+    dataset: Dict[str, Any] = None
+
+    @validator("model_class", always=True)
+    def task_or_model_class_required(cls, v, values):
+        if values["model_name"]:
+            if not v and not values.get("task", None):
+                raise ValueError("Either task or model_class must be specified")
+            return v
+
+
+class HFModelBase(torch.nn.Module):
+    """Base class for all HF models"""
```

Comment:
This is not a HFModelBase, it is just a PyTorchNetwork

We have used the OliveModelBase and its derived classes to handle ONNX, PT etc., models and their weights on the disk.

Reply:
Other alternative is to rename OliveModelBase as OliveModelWrapper (and its derived classes accordingly) and use PyTorchModel here instead of HFModelBase.

Reply:
Let me revisit this if we actually need a model class for HF model in Olive. If only conversion, we may not need a model class here.

Reply:
I agree with @devang-ml , the HFModelBase is confusing.

Reply:
Removed HFModelBase class.